### PR TITLE
change: Migrate credentials to use google-auth-library

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Migrate Application default credentials, Service account credentials, Refresh token credentials, and Compute engine credentials to use `google-auth-library`

Resolves: #1377

### CODE REVIEW

**Review by jonathanedey — COMMENTED** [review]
Looks good so far! Left one comment.

**Review by hsubox76 — APPROVED** [review]

**Comment by foxrafa** [comment]
I submitted a fix for ADC not working for human accounts here #2553

**Comment by VitaliAndrushkevich** [comment]
Could you please provide an update on the status of this PR?
CC: @lahirumaramba

**Comment by KemingHe** [comment]
@lahirumaramba @hsubox76 Hello there, as a `firebase admin sdk` user, I much look forward to this PR getting through, could you please give us a status update? Thank you.

File: src/app/credential-internal.ts [hunk 1]
```diff
@@ -507,122 +420,62 @@ export function getApplicationDefault(httpAgent?: Agent): Credential {
...
-  try {
-    return JSON.parse(fileText);
-  } catch (error) {
-    throw new FirebaseAppError(
-      AppErrorCodes.INVALID_CREDENTIAL,
-      'Failed to parse contents of the credentials file as an object: ' + error,
-    );
+  return {
+    ...credentials,
+    access_token: accessToken,
```
[the remark this answers is not in the document]
Comment (lahirumaramba) [reply src/app/credential-internal.ts:475]: Yeah I am keeping the shape of the internal `GoogleOAuthAccessToken` type here

**Pull request merged**